app/modules/perf.c

```diff
@@ -39,7 +38,11 @@ static void ICACHE_RAM_ATTR hw_timer_cb(os_param_t p)
   uint32_t stackaddr;
 
   if (data) {
-    uint32_t pc = *(&stackaddr + data->pc_offset);
+    uint32_t pc;
+    asm (
+      "rsr   %0, EPC1;"    /* read out the EPC */
```

Comment:
The pragmatic answer is that it works. The fuller answer is that the timer interrupt option has three compare registers which can each generate interrupts (each of a particular level). It appears that the SDK initializes the timer that the hw_timer uses to use interrupt 1 and hence EPC1 is the right register to use.